File: src/com/github/hmiyado/repository/entries/EntryRepository.kt
package com.github.hmiyado.repository.entries

import com.github.hmiyado.model.Entry

interface EntryRepository {
    fun getEntries(): List<Entry>

    fun createEntry(title: String, body: String): Entry

    fun getEntry(serialNumber: Long): Entry?

    fun updateEntry(serialNumber: Long, title: String? = null, body: String? = null): Entry?

    fun deleteEntry(serialNumber: Long)
}

File: api/entry/entryRepository.ts
import KottageClient from '../kottageClient'

const createEntry = async (title: string, body: string) => {
  return KottageClient.entriesPost({
    entriesPostRequest: {
      title,
      body,
    },
  })
}

const getEntries = async (limit: number = 20, offset: number = 0) => {
  return KottageClient.entriesGet({
    limit,
    offset,
  })
}

const fetchEntry = async (serialNumber: number) => {
  return KottageClient.entriesSerialNumberGet({ serialNumber })
}

const createComment = async (
  serialNumber: number,
  name: string,
  body: string
) => {
  return KottageClient.entriesSerialNumberCommentsPost({
    serialNumber,
    commentsPostRequest: {
      name,
      body,
    },
  })
}

const EntryRepository = {
  createEntry,
  getEntries,
  fetchEntry,
  createComment,
}

export default EntryRepository
